### PR TITLE
Supplemental Claims | Show alert when missing DoB or SSN

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- *(Summarize the changes that have been made to the platform)*
  > - Update Verify identity alert to use a green lock icon ([continue type alert per design](https://www.sketch.com/s/610156b6-f281-4497-81f3-64454fc72156/p/5E24F540-2558-488C-BD6D-823F4A2F5607/canvas))
  > - Add error alert shown when the user is LOA3, but missing their DoB or SSN
  > - Update unit tests
  > - Update mock data for e2e tests
- *(If bug, how to reproduce)*
- *(What is the solution, why is this the solution)*
- *(Which team do you work for, does your team own the maintainance of this component?)*
  > Benefits Team 1
- *(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)*

## Related issue(s)

[#48159](https://github.com/department-of-veterans-affairs/va.gov-team/issues/48159) - ticket is for the Veteran info page, but the alert is shown on the introduction page

## Testing done

- *Describe what the old behavior was prior to the change*
  > - An alert would show for non-LOA3 verification, but was a warning alert
  > - No check for SSN or DoB; user was allowed to start the form
- *Describe the steps required to verify your changes are working as expected*
  > `user` api call includes a `profile.claims.appeals` boolean which indicates that the user is both LOA3 and has a SSN
  > `user` is not available in the local environment, so the check is skipped while running locally
  > Run PR review instance and disable React component properties: `isVerified`, `hasDob` and `canApply` (appeals flag)
- *Describe the tests completed and the results*
  > Updated unit test
  > Updated e2e mock data

### After

<details><summary>LOA3 verified with DoB and SSN</summary>

Start your claim action link shown</details>

<details><summary>Save in progress content visible when LOA3, has SSN and has DoB</summary>

save-in-progress continue or start a new application buttons shown</details>

<details><summary>Need identity verification alert</summary>

verify your identity alert with link to verify page</details>

<details><summary>LOA3, but missing SSN and DoB</summary>

Missing SSN & DOB alert message</details>

<details><summary>LOA3, but missing SSN</summary>

Missing SSN alert message</details>

<details><summary>LOA3, but missing DoB</summary>

Missing DOB alert message</details>

## What areas of the site does it impact?

Supplemental Claims form (not yet in production)

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by data-doge — APPROVED** [review]
Feel free to merge. Left questions to satisfy my own curiosity.

File: src/applications/appeals/995/containers/IntroductionPage.jsx [hunk 1]
```diff
@@ -170,11 +189,16 @@ IntroductionPage.propTypes = {
...
-  };
-}
+const mapStateToProps = state => ({
+  loggedIn: isLoggedIn(state),
+  // Verified LOA3?
+  isVerified: selectProfile(state)?.verified || false,
+  canApply:
+    // profile.claims.appeals indicates that the Veteran can apply for an
+    // appeal (is LOA3 AND has a SSN), but it is undefined locally :(
+    // see vets-api/app/policies/appeals_policy.rb
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/995/containers/IntroductionPage.jsx:199]: Oh what that means to say is that in our local environment, the prefill (which includes the last 4 of the SSN) isn't available until after the Veteran starts the form.

File: src/applications/appeals/995/components/NeedsToVerify.jsx [hunk 2]
```diff
@@ -1,21 +1,44 @@
...
-      <a href={`/verify?next=${pathname}`} className="verify-link">
-        Verify your identity to start your request
-      </a>
-    </p>
-  </va-alert>
-);
+import recordEvent from 'platform/monitoring/record-event';
+
+const NeedsToVerify = ({ pathname }) => {
+  recordEvent({
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/995/components/NeedsToVerify.jsx:7]: Yes, the web-components have GA analytics built-in. Including this extra event is probably overkill. I can remove it.

File: src/applications/appeals/995/components/MissingInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import PropTypes from 'prop-types';
+
+import recordEvent from 'platform/monitoring/record-event';
+
+const MissingInfo = ({ hasSsn, hasDob }) => {
+  const hasSsnOrDob = [
+    hasSsn ? '' : 'Social Security number',
+    hasDob ? '' : 'date of birth',
+  ].filter(Boolean);
+  const missing = hasSsnOrDob.join(' and ');
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/components/MissingInfo.jsx:11]: If DOB and SSN do exist will `missing` still join the empty strings?

File: src/applications/appeals/995/components/MissingInfo.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  recordEvent({
+    event: 'visible-alert-box',
+    'alert-box-type': 'warning',
+    'alert-box-heading': 'We’re missing some of your personal information',
+    'error-key': 'missing_ssb_or_dob',
```
Comment (rocio-desantiago) [inline src/applications/appeals/995/components/MissingInfo.jsx:17]: small typo: `'missing_ssn_or_dob'`

Comment (Mottie) [reply src/applications/appeals/995/components/MissingInfo.jsx]: Oops, thanks! Nice find!

File: src/applications/appeals/995/components/MissingInfo.jsx (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/995/components/MissingInfo.jsx:11]: The `join` won't work with an array with 1 or less entry. So we will end up with an empty string

**Pull request merged**